azurelinuxagent/common/protocol/extensions_goal_state_from_vm_settings.py

```diff
@@ -232,8 +236,96 @@ def _parse_agent_manifests(self, vm_settings):
+        #                     "publicSettings": "{\"GCS_AUTO_CONFIG\":true}"
+        #                 }
+        #             ]
+        #         },
+        #         {
+        #             "name": "Microsoft.CPlat.Core.RunCommandHandlerLinux",
+        #             "version": "1.2.0",
+        #             "location": "https://umsavbvncrpzbnxmxzmr.blob.core.windows.net/f4086d41-69f9-3103-78e0-8a2c7e789d0f/f4086d41-69f9-3103-78e0-8a2c7e789d0f_manifest.xml",
+        #             "failoverlocation": "https://umsajbjtqrb3zqjvgb2z.blob.core.windows.net/f4086d41-69f9-3103-78e0-8a2c7e789d0f/f4086d41-69f9-3103-78e0-8a2c7e789d0f_manifest.xml",
+        #             "additionalLocations": [
+        #                 "https://umsawqtlsshtn5v2nfgh.blob.core.windows.net/f4086d41-69f9-3103-78e0-8a2c7e789d0f/f4086d41-69f9-3103-78e0-8a2c7e789d0f_manifest.xml"
+        #             ],
+        #             "state": "enabled",
+        #             "autoUpgrade": true,
+        #             "runAsStartupTask": false,
+        #             "isJson": true,
+        #             "useExactVersion": true,
+        #             "settingsSeqNo": 0,
+        #             "isMultiConfig": true,
+        #             "settings": [
+        #                 {
+        #                     "publicSettings": "{\"source\":{\"script\":\"echo '4abb1e88-f349-41f8-8442-247d9fdfcac5'\"}}",
+        #                     "seqNo": 0,
+        #                     "extensionName": "MCExt1",
+        #                     "extensionState": "enabled"
+        #                 },
+        #                 {
+        #                     "publicSettings": "{\"source\":{\"script\":\"echo 'e865c9bc-a7b3-42c6-9a79-cfa98a1ee8b3'\"}}",
+        #                     "seqNo": 0,
+        #                     "extensionName": "MCExt2",
+        #                     "extensionState": "enabled"
+        #                 },
+        #                 {
+        #                     "publicSettings": "{\"source\":{\"script\":\"echo 'f923e416-0340-485c-9243-8b84fb9930c6'\"}}",
+        #                     "seqNo": 0,
+        #                     "extensionName": "MCExt3",
+        #                     "extensionState": "enabled"
+        #                 }
+        #             ]
+        #         }
+        #         ...
+        #     ]
+        #     ...
+        # }
+        extension_goal_states = vm_settings.get("extensionGoalStates")
+        if extension_goal_states is not None:
+            if not isinstance(extension_goal_states, list):
+                raise Exception("extension_goal_states should be an array")
+            for extension_gs in extension_goal_states:
+                extension = Extension()
```

Comment:
NIT: Looks like at this level of parsing we are calling these are `ext_handler` that's how it's defined in other places. Can you make it consistent?

Reply:
We should change the names from `ext_handler` to `extension` in the other file rather than changing it here, that would make the intension more clearer

Reply:
I posted an overall comment wrt additional renames.